web/components/recipes.templ

```diff
@@ -174,6 +174,34 @@ templ addRecipe() {
 					standard, the <a href="https://www.mastercook.com/" target="_blank" class="link">MasterCook</a> file
 					format, or as plain text files.
 				</p>
+				<p>
+					You can also download recipe schema files directly using the
+					<a
+						class="link tooltip"
+						data-tip="Simply drag this link to your bookmarks bar, and click the bookmark while on a recipe website.  If a recipe schema downloads successfully, you can import it here."
+						href="javascript:(function(){
+							const recipeCount = 0;
+							jsonSchemaScripts = ([...document.querySelectorAll('script[type=%22application/ld+json%22]')]);
+							for (const jsonSchemaScript of jsonSchemaScripts) {
+								const data = JSON.parse(jsonSchemaScript.innerHTML);
```

Comment:
Sometimes, the script[type='application/ld+json'] is an Array of RecipeSchema objects, e.g. https://www.allrecipes.com/recipe/231961/double-banana-nut-bread. 
To address these cases, we can verify whether `data` is an Array. If it is, we can then retrieve the first element from it.

For example,
```javascript
let data = JSON.parse(jsonSchemaScript.innerHTML);
if (Array.isArray(data) && data.length > 0) {
    data = data[0];
}
```

Reply:
I had no idea, but it's a great pick-up. I'll have a look at both these tomorrow morning :)

Reply:
Resolved, though the image isn't working... might be worth extracting an og:image or something too?

![Uploading image.png…]()

Reply:
Looks like the extracted JSON+LD data does have an image url embedded in the file, but it's not being fetched as part of the JSON+LD import process.

If you're happy for me to open a bug for that behaviour (and I'll have a crack at fixing some time soon), this is probably ok for a first cut?